### PR TITLE
Console feature to run current file

### PR DESCRIPTION
**Pull request by grondinjc**
The implementation of #65 with applied changes after review

### CODE REVIEW

File: src/app/python/core.py [hunk 1]
```diff
@@ -427,7 +598,7 @@ def _inner_task_apply_changes(self, path):
       self._logger.exception("EXCEPTION RAISED {0}\n{1}\n{2}".format(e[0], e[1], e[2]))
 
   @task_time(microseconds=1)
-  def _task_create_archive(self, path, caller, response):
```
Comment (grondinjc) [inline src/app/python/core.py:430]: was moved up

**Comment by grondinjc** [comment]
@vphilippon

File: src/server/ideController.py [hunk 2]
```diff
@@ -47,7 +47,30 @@ def create_tree_nodes_dict(nodes):
 
 def set_author_bool_in_dict(serialized_changes, user):
   for sc in serialized_changes:
-    sc['is_from_you'] = (sc['author'] == user)
+    sc['author'] = (sc['author'] == user)
```
Comment (vphilippon) [inline src/server/ideController.py:50]: The first occurence of 'author' key was changed to 'is_from_you', else the author name will be lost for further itteration in the user send loop.
(Ex: gives "if False == grondinjc" ...)
The correct line should be the original:
`sc['is_from_you'] = (sc['author'] == user)`

File: src/server/ideController.py [hunk 3]
```diff
@@ -257,10 +287,10 @@ def save(self):
...
                   'vers':    '<<File version>>',
                   'changes': [{
-                               'type':        '<<Type of edit (ins | del)>>',
-                               'pos':         '<<Position of edit>>',
-                               'content':     '<<Content of insert | Number of deletes>>'
-                               'is_from_you':  <<If the user is the author>>
+                               'type':    '<<Type of edit (ins | del)>>',
+                               'pos':     '<<Position of edit>>',
+                               'content': '<<Content of insert | Number of deletes>>'
+                               'author':   <<If the user is the author>>
```
Comment (vphilippon) [inline src/server/ideController.py:293]: Same as above, keep original
`is_from_you`

File: src/server/ideController.py [hunk 4]
```diff
@@ -425,10 +556,10 @@ def _save_callback(self, filename, changes, version, users):
...
```
Comment (vphilippon) [inline src/server/ideController.py:562]: Same as above, keep original
`is_from_you`

File: src/static/js/utils/representation.js [hunk 5]
```diff
@@ -25,11 +27,18 @@ function createExport(directory) { return { path: directory}; }
...
+// used for /ide/execstart
+function createProcessInit(filename, arguments) { return {file: filename, args: arguments}; }
+// used for /ide/execinput
+function createProcessInput(arguments) { return {data: arguments}; }
+// used for /ide/execkill
+function createProcessTermination() { return {}; }
 
-function ObjectAddChange(pos, content, is_from_you){
+
+function ObjectAddChange(pos, content, author){
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:38]: Keep original `is_from_you`, it's the new key

File: src/static/js/utils/representation.js [hunk 6]
```diff
@@ -25,11 +27,18 @@ function createExport(directory) { return { path: directory}; }
...
   this._pos = pos;
   this._content = content;
-  this._is_from_you = is_from_you; // From server only
+  this._author = author; // From server only
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:41]: Same, keep original
`this._is_from_you = is_from_you;`

File: src/static/js/utils/representation.js [hunk 7]
```diff
@@ -38,22 +47,24 @@ ObjectAddChange.prototype.applyOnText = function(text){
   return text.insert(this._content, this._pos);
 };
 ObjectAddChange.prototype.applyOnPos = function(pos){
-  if(!this._is_from_you){
+  if(!this._author){
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:50]: `this._is_from_you`

File: src/static/js/utils/representation.js [hunk 8]
```diff
@@ -38,22 +47,24 @@ ObjectAddChange.prototype.applyOnText = function(text){
...
     pos += (this._pos <= pos ? this.size() : 0);
   }
   return pos;
 };
 ObjectAddChange.prototype.updatePos = function(modObject){
-  this._pos = modObject.applyOnPos(this._pos);
+  if(!this._author){
```
Comment (vphilippon) [inline src/static/js/utils/representation.js]: `this._is_from_you`

File: src/static/js/utils/representation.js [hunk 9]
```diff
@@ -38,22 +47,24 @@ ObjectAddChange.prototype.applyOnText = function(text){
...
+    this._pos = modObject.applyOnPos(this._pos);
+  }
 };
 ObjectAddChange.prototype.serialize = function(){
   return createAddModif(this._content, this._pos);
 };
 
-function ObjectRemoveChange(pos, count, is_from_you){
+function ObjectRemoveChange(pos, count, author){
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:64]: Same, keep original`is_from_you;`

File: src/static/js/utils/representation.js [hunk 10]
```diff
@@ -38,22 +47,24 @@ ObjectAddChange.prototype.applyOnText = function(text){
...
   this._pos = pos;
   this._count = count;
-  this._is_from_you = is_from_you; // From server only
+  this._author = author; // From server only
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:67]: `this._is_from_you = is_from_you;`

File: src/static/js/utils/representation.js [hunk 11]
```diff
@@ -62,13 +73,15 @@ ObjectRemoveChange.prototype.applyOnText = function(text){
   return text.cutFrom(this._pos, this._count);
 };
 ObjectRemoveChange.prototype.applyOnPos = function(pos){
-  if(!this._is_from_you){
+  if(!this._author){
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:76]: `this._is_from_you`

File: src/static/js/utils/representation.js [hunk 12]
```diff
@@ -62,13 +73,15 @@ ObjectRemoveChange.prototype.applyOnText = function(text){
...
     pos -= (this._pos <= pos ? this.size() : 0);
   }
   return pos;
 };
 ObjectRemoveChange.prototype.updatePos = function(modObject){
-  this._pos = modObject.applyOnPos(this._pos);
+  if(!this._author){
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:82]: Keep original, test in done in `applyOnPos`

File: src/static/js/utils/representation.js (hunk 8 above)
Comment (vphilippon) [inline src/static/js/utils/representation.js:56]: Keep original, test in done in `applyOnPos`

File: src/static/js/utils/representation.js [hunk 13]
```diff
@@ -78,9 +91,9 @@ ObjectRemoveChange.prototype.serialize = function(){
 function ObjectChangeFactory(changeObjJSON){
   return changeObjJSON.type == CHANGE_ADD_TYPE ?
     new ObjectAddChange(changeObjJSON.pos, changeObjJSON.content,
-                        changeObjJSON.is_from_you):
+                        changeObjJSON.author):
```
Comment (vphilippon) [inline src/static/js/utils/representation.js:94]: `is_from_you`

**Comment by vphilippon** [comment]
@grondinjc I tagged a bunch of stuff, but overall, it's:
Keep the 'is_from_you' change, that's an important fix.

File: src/app/python/core.py [hunk 14]
```diff
@@ -439,29 +610,31 @@ def _task_create_archive(self, path, caller, response):
     @param caller: The user name
     @param response: Synchrone helper on which response needs to be written
     """
```
Comment (maym2104) [inline src/app/python/core.py:612]: Update comment.

**Comment by vphilippon** [comment]
I'll fix in a fast PR

**Pull request merged**